.github/workflows/deploy_prod.yaml

```diff
@@ -6,6 +6,69 @@ on:
       - main
 
 jobs:
+  migrate_database:
+    runs-on: ubuntu-latest
+
+    env:
+      CLOUDSDK_CORE_DISABLE_PROMPTS: 1
+
+    steps:
+      - name: "checkout repo"
+        uses: actions/checkout@v2
+
+      - name: "setup Python"
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: "use pip cache"
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ hashFiles('**/requirements*.txt') }}
+          restore-keys: |
+            pip-
+
+      - name: "install dependencies"
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt -r requirements-dev.txt
+
+      - name: "authenticate google cloud"
+        id: "auth"
+        uses: "google-github-actions/auth@v1"
+        with:
+          service_account: ${{ secrets.GCP_DEPLOY_KEY }}
+
+      - name: "gcloud setup"
+        uses: google-github-actions/setup-gcloud@v1
+        with:
+          project_id: curator-348003
+          install_components: "cloud_sql_proxy"
+
+      - name: "gcloud secrets"
+        id: "secrets"
+        uses: "google-github-actions/get-secretmanager-secrets@v1"
+        with:
+          secrets: |-
+            secret_key:curator-348003/django-secret-key-prod/latest
+            db_password:curator-348003/postgres-password-prod/latest
+
+      - name: "apply database migrations"
+        env:
+          DJANGO_SETTINGS_MODULE: curation_portal.settings.base
+          DB_ENGINE: django.db.backends.postgresql
+          DB_HOST: localhost
+          DB_PORT: 5432
+          DB_DATABASE: curator-prod
+          DB_USER: curator-prod
+          DB_PASSWORD: "${{ steps.secrets.outputs.db_password }}"
+          SECRET_KEY: "${{ steps.secrets.outputs.secret_key }}"
+        run: |
+          ./cloud-sql-proxy \
+            --port 5432 curator-348003:australia-southeast1:curator-postgres &
+            python manage.py migrate
```

Comment:
Instead of launching a proxy and doing the migration on the GitHub CI instance itself, can we instead use a Cloud Run job (https://cloud.google.com/run/docs/create-jobs)?

This way, the invocation should look much more similar to the "build and deploy" step below, with everything running in the same Docker container, on GCP.

Reply:
Creating a cloud run job requires a pre-built image, so I've added a build step into the workflow. This image is then used in the `migrate` and `deploy` jobs. This will require activation of the Artefact Repository API, and an update to the service account so that pushing to this repository is allowed.

Reply:
Excellent, thanks! I've created that AR repository and updated the permissions.